### PR TITLE
PHPC-317: Ensure "sort" query option serializes as BSON object

### PR DESCRIPTION
**Pull request by jmikola**
https://jira.mongodb.org/browse/PHPC-317

### CODE REVIEW

File: php_phongo.c [hunk 1]
```diff
@@ -342,7 +341,6 @@ bool phongo_query_init(php_phongo_query_t *query, zval *filter, zval *options TS
 				return false;
 			}
 
-			convert_to_array_ex(&projection);
 			query->selector = bson_new();
```
Comment (bjori) [inline php_phongo.c]: How is this related to $sort ?
Is it similar issue, in which case, should it not be convert_to_object_ex() then?

**Comment by bjori** [comment]
I think you need to kick travis to see if it likes it

File: php_phongo.c (hunk 1 above)
Comment (jmikola) [inline php_phongo.c]: The projection and query document (i.e. what we merge modifiers into above) didn't need this treatment since they are top-level documents and you encode them as such later on. Sort was the unique case where it gets stuffed in an `$orderby` field and could be encoded as an array inadvertently.

I do mean to split these changes up into separate commits, but the `convert_to_array_ex()` deletions here and for modifiers are basically just removing dead code.

**Comment by jmikola** [comment]
Travis looks good. This will need a test for sorting on `{ "0": 1 }` and `{ "0": 1, "1": 1 }` to test for proper serialization, though.

File: php_phongo.c (hunk 1 above)
Comment (bjori) [inline php_phongo.c]: oki lgtm

**Comment by jmikola** [comment]
Updated commit to test for `{ "0" : 1 }`, which was all that's really needed. We can verify that the serialization is correct through Query's debug handler.

Additionally, 3.0.x seems to ignore the `$orderby` value if it's not an object. Testing that the documents come back in an order other than their natural (inserted) order helps to test that the server is actually taking our `$orderby` value into account.

**Comment by bjori** [comment]
lgtm
